### ISSUE TITLE
[FEATURE] Add borderType option to kornia.filters like OpenCV

### ISSUE THREAD

**Issue by S-aiueo32** [arraiyopensource/kornia#170]
`kornia.filters` operations have no option like `borderType` on OpenCV.
In my experience, `borderType` is the most critical argument in the arguments that is not in `kornia.filters`.
To reproduce it, I suggest the changes like below:
```python
class BoxBlur(nn.Module):
    def __init__(self, kernel_size: Tuple[int, int], border_type='zero': str) -> None:
        super(BoxBlur, self).__init__()
        self.kernel: torch.Tensor = _get_box_filter(kernel_size)
        self.padding: Tuple[int, int] = _compute_zero_padding(kernel_size)
        self.border_type = border_type

    def forward(self, input: torch.Tensor):  # type: ignore
        if not torch.is_tensor(input):
            raise TypeError("Input type is not a torch.Tensor. Got {}"
                            .format(type(input)))
        if not len(input.shape) == 4:
            raise ValueError("Invalid input shape, we expect BxCxHxW. Got: {}"
                             .format(input.shape))
        # prepare kernel
        b, c, h, w = input.shape
        tmp_kernel: torch.Tensor = self.kernel.to(input.device).to(input.dtype)
        kernel: torch.Tensor = tmp_kernel.repeat(c, 1, 1, 1)

        return conv2d(input, kernel, padding=self.padding, padding_mode=self.border_type, stride=1, groups=c)
```

**Comment by edgarriba** [comment]
@S-aiueo32 you are absolutely right. However, [`torch.conv2d`](https://pytorch.org/docs/stable/nn.html#id22) only implements `zeros` or `circular` padding modes. To address this issue, as @ducha-aiki suggested in 
https://github.com/arraiyopensource/kornia/pull/168, we need to manually pad the input using `torch.pad` which already implement more advanced padding techniques. In order to generalize the other filtering operators, I will implement this strategy in a [`filter2d`](https://docs.opencv.org/master/d4/d86/group__imgproc__filter.html#ga27c049795ce870216ddfb366086b5a04) operator which is BTW already implemented in OpenCV.

**Comment by S-aiueo32** [comment]
I had noticed that `torch.conv2d` has a poor option for padding after my post.
So... how long will it take to release `filter2d` operation?
If it takes a lot, I think it is needed to make ad hoc changes for `border_type` using pure `pad` operation.
```python
from torch.nn.functional import pad

def forward(self, input: torch.Tensor):  # type: ignore
    if not torch.is_tensor(input):
        raise TypeError("Input type is not a torch.Tensor. Got {}"
                        .format(type(input)))
    if not len(input.shape) == 4:
        raise ValueError("Invalid input shape, we expect BxCxHxW. Got: {}"
                            .format(input.shape))
    # prepare kernel
    b, c, h, w = input.shape
    tmp_kernel: torch.Tensor = self.kernel.to(input.device).to(input.dtype)
    kernel: torch.Tensor = tmp_kernel.repeat(c, 1, 1, 1)
    padded: torch.Tensor = pad(input, [kernel[0], kernel[0], kernel[1]], kernel[1], mode=self.border_type)
    return conv2d(padded, kernel, padding=0, stride=1, groups=c)
```

**Comment by edgarriba** [comment]
@S- aiueo32 check this out . Already merged https://github.com/arraiyopensource/kornia/pull/171

**Comment by S-aiueo32** [comment]
Ohhhh... I missed it.
Thank you for the great job.

**Comment by edgarriba** [comment]
Yeap, it's something we needed for a while. I've seen you implemented YCbCr, are you planning to send a PR? The is a lot of color stuff to be done 😆

**Comment by S-aiueo32** [comment]
@edgarriba Yeah, I'll throw it.
